### DISCUSSION TITLE
Prompt templates

### DISCUSSION THREAD

**Discussion by fargot132** [danny-avila/LibreChat#632] [Feature Requests & Suggestions]
Hello
Great work !
It will be very helpful to implement prompt templates as in https://github.com/mckaywrigley/chatbot-ui
Unfortunately, this project is abandoned, but the idea of prompt templates is very interesting.
The next step of improvement may be the sharing of prompt templates between users.

**Comment by danny-avila** [comment]
Thank you! There are already "presets" which have a similar function, but don't have the filling in of variables. They are actively being shared in our discord: https://discord.gg/nXdfrht9DF

This is not to say this wouldn't be implemented, it's a great idea, and I would like people to have the option to fill them out in-line in the textarea box, which is an improvement over chatbot-ui. I'm wrapping up ChatGPT plugins and Anthropic support, then diving into file indexing/retrieval as my priorities in the roadmap.

A PR for this would be very welcome.

**Comment by secondtruth** [comment]
I think this would be a great feature.

Presets are a pretty nice feature but lack the full flexibility of "saved prompts" / "prompt templates", which in contrast to presets:
- are in principle always available to every model without tediously defining them for each model individually.
- can also be used in the middle of a conversation instead of just at the beginning.

A library of saved prompts could also come in handy to fill in the System Prompt field while creating presets or assistants. Instead of manually entering or copy-pasting the prompt, one could click on an "Import" button and choose a prompt from the library.

Further improvements for this hypothetical Prompts Library could include:

- Ability to save prompts from past chats in the library
- Ability to import prompts into the library from a community source like https://github.com/f/awesome-chatgpt-prompts

**Comment by TheRealAlexV** [comment]
I would also fully support this. I switched from using chatgpt plus coupled with superpowers gpt to librechat and really miss having a library of popular promts with variable expansion. I still hop over to chatgpt from time to time because of the library of promts that I commonly use there. The presets feature is very confusing for this use case.

**Reply by danny-avila** [reply]
Prompts are already live

**Reply by danny-avila** [reply]
There are a couple of remaining features, track here: https://github.com/danny-avila/LibreChat/issues/3023

**Reply by TheRealAlexV** [reply]
> Prompts are already live

How do I go about accessing that feature? I believe I'm running the latest version of LibreChat `LibreChat v0.7.4-rc1`.

**Reply by danny-avila** [reply]
It’s in the side panel, do you not see this? How did you update?

**Reply by danny-avila** [reply]
<img width="414" alt="image" src="https://github.com/danny-avila/LibreChat/assets/110412045/cee6b7d0-883a-4551-bb00-5467f9fdf4b7">

**Reply by TheRealAlexV** [reply]
> It’s in the side panel, do you not see this? How did you update?

I do not see that in my side panel:
![image](https://github.com/danny-avila/LibreChat/assets/815793/53b3e8b9-3a0b-42c6-a56d-e51d87389ddb)

To update I:
1. `docker-compose down`
2. `git pull`
3. `docker-compose pull`
4. `docker-compose up -d`

**Reply by danny-avila** [reply]
Are you pointing to a specific image? It looks like you are not on the latest version just from the footer text alone

**Reply by danny-avila** [reply]
It’s not part of the rc1 tag, it’s on latest commits to the main branch so the latest tag

**Reply by TheRealAlexV** [reply]
Thank you Danny! It was the image tag. My docker override was using image: ghcr.io/danny-avila/librechat:latest [which is what the .example file gives you as a template](https://github.com/danny-avila/LibreChat/blob/156c52e293573b7e3990ab2137f14ab5a857e95b/docker-compose.override.yml.example#L22) which makes sense since the documentation should probably always point to the stable non-dev version. Might be good to note how to switch to dev though(if I didn't miss it in the docs somewhere).

I commented out the line in my override so it's using the image from the parent docker compose file which is image: ghcr.io/danny-avila/librechat-dev:latest.
